packages/tiled/lib/src/common/property.dart

```diff
@@ -33,14 +33,21 @@ class Property {
 }
 
 extension PropertiesParser on Parser {
-  List<Property> getProperties() {
-    return formatSpecificParsing(
+  Map<String, Property> getProperties() {
```

Comment:
Can we minimize the breaking change by adding a new `getPropertiesMap()` that returns `Map<String, Property>` and making `getProperties()` return the `values` iterable  from the map?

Reply:
I don't think anyone uses this `getProperties` method directly, but rather the `properties` public variable in `Tile` `Object` `Layer` etc.

We definitely could change the API to:
```dart
class Tile {
  ...

  Iterable<Property> get properties => propertiesByName.values;
  final Map<String, Property> propertiesByName;
}
```

But personally I think that's a worse API and there is really no value in an `Iterable<Property>` accessor, especially when you can get it via `.values`. 

I was under the impression that we're in early stages with this library and want to do large refactors to polish the API before 1.0 so that's why I was leaning towards the more aggressive refactor. But, I'm happy either way 🤷

Reply:
> I don't think anyone uses this `getProperties` method directly, but rather the `properties` public variable in `Tile` `Object` `Layer` etc.

I get what you mean, but this is still just an assumption. If it is a public API, it is always better to assume that it is used by someone. Also, even if `getProperties` is not used by anyone, we are still breaking `properties`.

> But personally I think that's a worse API and there is really no value in an `Iterable<Property>` accessor, especially when you can get it via `.values`.
> 
> I was under the impression that we're in early stages with this library and want to do large refactors to polish the API before 1.0 so that's why I was leaning towards the more aggressive refactor. But, I'm happy either way 🤷

@spydon can comment about plans for 1.0, but I feel making a breaking change and trying not to bump up the major version are conflicting goals here. If you make the changes non-breaking, it will automatically solve the versioning problem. A middle ground would be to let `getProperties` and `properties` work as they used to with a `@deprecated` tag saying that it will be removed in 1.0.

*I might have a hidden motive for making it non-breaking, because `properties` is used in my simple platformer series. It will be better if maintainers should what to do about this 😅

Reply:
It doesn't need to have a major bump when it is breaking and sub v1. I think that we can deprecate `getProperties`, but not remove it until we go to past v1, `properties` is more inline with the dart naming convention anyways.
So then your tutorials wouldn't break for a long time.

Poke me when you feel that a new tiled + flame_version should be released btw.

Reply:
So, keep `getProperties` and `properties` and add `getPropertiesByName` and `propertiesByName`?

To be clear `getProperties` and `properties` are two different things, `getProperties` is the function that parses the data which eventually gets set on the `properties` class level vars.

Reply:
Aah right, hard to review PRs properly on the phone. Should `getProperties` only be used internally, or is there a value in exposing that to the user? If it's not I would name it `parseProperties` and mark it as `@internal`.
And for `properties` I think that we should do a breaking change on the type then and not pollute the code with unnecessary members already.

Reply:
Sounds good, do you want me to rename ALL the parser methods? 

- Parser.getString
- Parser.getStringOrNull
- Parser.getInt
- Parser.getColor
- etc etc

Reply:
Not sure, I can take a look.

For the method naming, I think the classname makes it self explanatory so I'm inclined to leave it as is, what do you think?

Reply:
> For the method naming, I think the classname makes it self explanatory so I'm inclined to leave it as is, what do you think?

Agree, missed that it was on the parser class when I reviewed on mobile.